### DISCUSSION TITLE
Event for theme

### DISCUSSION THREAD

**Discussion by BoiteBeet** [laravel/nova-issues#5322] [Ideas & Feature Requests] [answered]
Add events in the theme switch so we can listen for them anywhere.
`Nova.$emit('theme-changed', 'dark')`

Right now I need to do an observer on the document.documentElement for the class 'dark' to be able to do it and it's really not the most performent.

It's particularily useful when developing components that needs some reloading for theme (i.e. some rich text fields like tinymce)

**Comment by gbrits** [comment]
Landed here because I needed the same thing.

**Comment by crynobone** [comment, marked as answer]
You should be able to do this in Nova 5.

```js
Nova.$on('nova-theme-switched', ({ theme }) => {
  // `theme` should be 'dark' or 'light'
})
```
